### PR TITLE
Consult the environment variable when setting the max users frames in code origin probes

### PR DESCRIPTION
**Pull request by evanchooly**
# What Does This Do

Update the code origin code to check for an environment variable when configuring the maximum number of user frames to include in a code origin probe's output.

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [DEBUG-2727]

[DEBUG-2727]: https://datadoghq.atlassian.net/browse/DEBUG-2727?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/CodeOriginProbe.java [hunk 1]
```diff
@@ -26,6 +26,8 @@
 public class CodeOriginProbe extends LogProbe implements ForceMethodInstrumentation {
   private static final Logger LOGGER = LoggerFactory.getLogger(CodeOriginProbe.class);
 
+  public static int MAX_FRAMES = 8;
```
Comment (jpbempel) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/CodeOriginProbe.java:29]: nit: naming suggests this is constant which is not. confusing.

Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/CodeOriginProbe.java:29]: putting this limit in public static is problmatic. I would pass it to the CodeOriginProbe Constructor when the probe is created. This is the DefaultCodeOriginRecorder that should maintain this limit (reading from config and pass it to the probes)

Comment (evanchooly) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/CodeOriginProbe.java:29]: eh.  the fields are DebuggerContext aren't constant either.  😄  I'm not opposed to making it a ctor parameter, though.  Might make its provenance a little less murky.

Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/CodeOriginProbe.java]: Sorry I was not explicit enough, I meant the capital casing.

**Review by jpbempel — APPROVED** [review]

**Pull request merged**